src/lib.rs

```diff
@@ -391,10 +391,10 @@ impl LightningNode {
     }
 
     pub fn get_node_info(&self) -> Result<NodeInfo> {
-        let node_state: NodeState = self.sdk.node_info().map_err(|e| RuntimeError {
-            code: RuntimeErrorCode::NodeUnavailable,
-            msg: e.to_string(),
-        })?;
+        let node_state = self
+            .sdk
+            .node_info()
+            .map_to_permanent_failure("Failed to read node info")?;
```

Comment:
Why permanent failure? The SDK seems to return `ServiceErrorCode::NotReady` in the version that is being worked on. This may not indicate a permanent problem but a runtime one instead.

Reply:
Maybe if it isn't persisted, it means it will be persisted soon, so still not a permanent failure. I would simply keep this as the `NodeUnavailable` runtime error. 

I don't think we should be looking into implementation of dependencies to figure out what errors to throw. That can easily lead to issues because the dependency implementation can change.